test/test_example.py

```diff
@@ -31,3 +31,16 @@ def test_errors(src_yaml):
     with pytest.raises(Exception) as exception:
         command.compile(ns)
     assert expected in str(exception.value)
+
+
+@pytest.mark.parametrize("src_yaml",
+                         [str(path) for path in (pathlib.Path(__file__).parent / "data/log_fatal").glob("*.yaml")])
```

Comment:
Silly question, why do we have a `test/data/error` and `test/data/log_fatal` dir? What is the difference, why do we need two?

Reply:
at the most upper call there is no exception - only ret == 1 and a fatal log

Reply:
Why would we do a different user experience, why would we use different code patterns? Are the errors actually different to justify this or is there a different reason? IMHO consistency in both code and user experience is important so either we use the same pattern(s) or if we don't we should at least have a code comment explaining why they diverge. (sorry I hope this not come accross too nitpicky/weird)

Reply:
Sorry, I may have been a bit terse and I see we already mixing some validation that raises exceptions and some that writes log messages  - I guess what I try to get to is this is maybe a good opportunity to step back and see if we can unify the error reporting to use the same pattern(s) :)